src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -264,7 +267,21 @@ public boolean isTopK()
     @VisibleForTesting
     public Index.Searcher indexSearcher()
     {
-        return indexQueryPlan == null ? null : indexQueryPlan.searcherFor(this);
+        if (indexQueryPlan == null)
```

Comment:
I did not see tests that show that things look ok if we have queries involving non-sai indexes. Can you add some, please?

Reply:
Added, and also for regular queries.

Reply:
I've added an artificial test for the sake of increasing coverage from the current 96.6%.